### PR TITLE
Add logging to `FileCache::tryReserve`

### PR DESCRIPTION
**Pull request by yakov-olkhovskiy**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Add logging to `FileCache::tryReserve`.
Flaky test investigation #66683.

cc @kssenii 

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit ea9a8b88120f357d2a545c7c2aef209d866e7b9f with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/66794/ea9a8b88120f357d2a545c7c2aef209d866e7b9f/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/66792/42ba3229b5484ebe29f4191cd5672433ba1f6588/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/Interpreters/Cache/FileCache.cpp [hunk 1]
```diff
@@ -889,6 +896,9 @@ bool FileCache::tryReserve(
     if (!main_priority->collectCandidatesForEviction(
             size, required_elements_num, reserve_stat, eviction_candidates, queue_iterator, user.user_id, cache_lock))
     {
+        LOG_DEBUG(log, "Collection candidates for eviction from cache is unsuccessful, "
```
Comment (kssenii) [inline src/Interpreters/Cache/FileCache.cpp:899]: I do not think it is worth it, this will pollute the logs unnecessarily.

Comment (yakov-olkhovskiy) [reply src/Interpreters/Cache/FileCache.cpp:899]: Is this about this specific LOG_DEBUG or the whole PR?

Comment (alexey-milovidov) [reply src/Interpreters/Cache/FileCache.cpp:899]: @yakov-olkhovskiy, replace with `LOG_TEST` and it will be good to merge.

Comment (yakov-olkhovskiy) [reply src/Interpreters/Cache/FileCache.cpp:899]: @alexey-milovidov we need this diagnostics in CI runs (at least temporarily - until we figure out the problem) - and CI as I remember don't run at TEST log level. If only this particular log then maybe ok - we will be able to figure it out by elimination.
@kssenii is it ok to convert only this one to LOG_TEST and left others at LOG_DEBUG?

Comment (kssenii) [inline src/Interpreters/Cache/FileCache.cpp:899]: Having any logging here would be too verbose while this log itself does not represent much benefit for debugging.

**Comment by alexey-milovidov** [comment]
FYI, the minimum level that is enabled in production is LOG_TRACE, not LOG_DEBUG.

**Comment by yakov-olkhovskiy** [comment]
@alexey-milovidov @kssenii if we can't add diagnostics which would be enabled in tests on CI then I don't see a point in this PR - failures in the test which I'm trying to debug are relatively rare. Then let's try to proceed with Kseniia's initial suggestion to increase `filesystem_cache_reserve_space_wait_lock_timeout_milliseconds` and see if test will stop failing.
I'm closing this PR now.

**Pull request closed without merging**